Firestore/core/src/firebase/firestore/util/objc_class.h

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2019 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_UTIL_OBJC_CLASS_H_
+#define FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_UTIL_OBJC_CLASS_H_
+
+// The OBJC_CLASS macro defines a forward declaration for an Objective-C class
+// that's compatible both with Objective-C++ and regular C++. It's useful in
+// headers that reference Objective-C types as members of C++ classes that must
+// be usable from straight C++.
+//
+// Care must be taken to not use the forward declaration (even implicitly) in
```

Comment:
I'm not sure I understand this part of the comment. Perhaps _"forward declaration"_ should be replaced with _"inline definitions"_? Also, from our offline discussion, it seemed like the underlying issue is breaking ODR (by having e.g. two different versions of the implicit destructor). Please correct me if I'm misremembering; otherwise, I'd suggest adding a bit about ODR and either omitting the ARC bit or integrating it into the ODR context.